docs/Binding-and-project-context-evaluation.md

```diff
@@ -21,14 +21,14 @@ The priority of the sources are following:
 The higher value indicates higher priority.
 
 
-|Name|Description|
-|---|---|
-|`type`|`bind`|
-|`binding`| Mandatory. The name of the source and parameter in the source to take the value from. The syntax follows: `<source prefix>:<parameter name>`.|
-|`replaces`|The text to be replaced by the symbol value in the template files content.|
-|`fileRename`|The portion of template filenames to be replaced by the symbol value.| 	 
-|`defaultValue`|The value assigned to the symbol if no value was provided from external source(s).|
-
+|Name|Description|Mandatory|
+|---|---|---|
+|`type`|`bind`|yes|
+|`binding`| Mandatory. The name of the source and parameter in the source to take the value from. The syntax follows: `<source prefix>:<parameter name>`.|yes|
+|`replaces`|The text to be replaced by the symbol value in the template files content.|no|
+|`fileRename`|The portion of template filenames to be replaced by the symbol value.|no| 	 
+|`defaultValue`|The value assigned to the symbol if no value was provided from external source(s). Recommended to be used when `replaces` and/or `fileRename` is used. In case default value is not specified and no value was provided from external source(s) the replacement won't be performed. |no|
+|`dataType`|The value assigned to the symbol if no value was provided from external source(s). Allowed values are: "bool", "float", "int", "hex", "text", "string". If not specified, the value type will be inferred. In case the type of values might be ambiguous, consider specifying the desired datatype for processing. In case the conversion of value to the type fails, the symbol will be skipped from  further processing. |no|
```

Comment:
Should we link to the symbol data types section https://github.com/dotnet/templating/blob/main/docs/Reference-for-template.json.md#parameter-symbol?

It has a bit more detail. The only difference is the support for choice type .... thinking about it a bit - I gues we can and should support it for bind symbols as well What do you think?

Reply:
I agree. Not only for `bind` symbols, but for all other cases where we support `dataType`.

Reply:
Created: https://github.com/dotnet/templating/issues/5828